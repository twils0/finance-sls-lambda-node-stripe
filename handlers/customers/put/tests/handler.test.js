const { testAuth } = require('../../../../functions/aws/testAuth');
const { loadStripe } = require('../../../../functions/stripe/loadStripe');
const { testUser } = require('../../../../functions/aws/testUser');
const { testCoupon } = require('../../../../functions/stripe/testCoupon');
const { getCustomer } = require('../../../../functions/stripe/getCustomer');
const { updateCustomer } = require('../../../../functions/stripe/updateCustomer');
const { updateSubscription } = require('../../../../functions/stripe/updateSubscription');
const { deleteDiscount } = require('../../../../functions/stripe/deleteDiscount');
const { deleteSubscription } = require('../../../../functions/stripe/deleteSubscription');
const { addSubscription } = require('../../../../functions/stripe/addSubscription');

const { put } = require('../handler');

global.console.log = jest.fn();
jest.mock('../../../../functions/aws/testAuth', () => ({
  testAuth: jest.fn(),
}));
jest.mock('../../../../functions/stripe/loadStripe', () => ({
  loadStripe: jest.fn(),
}));
jest.mock('../../../../functions/aws/testUser', () => ({
  testUser: jest.fn(),
}));
jest.mock('../../../../functions/stripe/testCoupon', () => ({
  testCoupon: jest.fn(),
}));
jest.mock('../../../../functions/stripe/getCustomer', () => ({
  getCustomer: jest.fn(),
}));
jest.mock('../../../../functions/stripe/updateCustomer', () => ({
  updateCustomer: jest.fn(),
}));
jest.mock('../../../../functions/stripe/updateSubscription', () => ({
  updateSubscription: jest.fn(),
}));
jest.mock('../../../../functions/stripe/deleteDiscount', () => ({
  deleteDiscount: jest.fn(),
}));
jest.mock('../../../../functions/stripe/deleteSubscription', () => ({
  deleteSubscription: jest.fn(),
}));
jest.mock('../../../../functions/stripe/addSubscription', () => ({
  addSubscription: jest.fn(),
}));
jest.mock('../../../../functions/aws/sns/publishSNS', () => ({
  publishSNS: jest.fn(),
}));
jest.mock('../../../../functions/errorResponse', () => ({
  errorResponse: jest.fn(error => error),
}));

const accessToken = 'testAccessToken';
const customerId = 'testCustomerID';
const sourceId = 'testSourceID';
const subscriptionId = 'testSubscriptionID';
const subscriptionItemId = 'testSubscriptionItemID';
const couponId = 'testCouponID';

const token = { test: 'testToken' };
const promoCode = 'testPromoCode';
const plan = 'basic_plan';
const email = 'test@test.com';
const name = 'testName';
const phone = '239-555-0000';

const event = {
  query: {
    accessToken,
  },
  cognitoPoolClaims: {
    stripe_customer_id: customerId,
  },
  body: {
    token,
    promoCode,
    plan,
    email,
    name,
    phone,
  },
};
const noPromoCodeEvent = {
  query: {
    accessToken,
  },
  cognitoPoolClaims: {
    stripe_customer_id: customerId,
  },
  body: {
    token,
    promoCode: null,
    plan,
    email,
    name,
    phone,
  },
};
const response = {
  status: 200,
  body: {
    message: 'success',
  },
};
const callback = jest.fn();

describe('handlers', () => {
  describe('users', () => {
    describe('put', () => {
      afterEach(() => {
        loadStripe.mockReset();
        testAuth.mockReset();
        testUser.mockReset();
        testCoupon.mockReset();
        getCustomer.mockReset();
        updateCustomer.mockReset();
        updateSubscription.mockReset();
        deleteDiscount.mockReset();
        deleteSubscription.mockReset();
        addSubscription.mockReset();
        global.console.log.mockReset();
        callback.mockReset();
      });

      it('fails and returns error, when testAuth throws error', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.reject(error));

        await put(event, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when loadStripe throws error', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.reject(error));

        await put(event, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when testUser throws error', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.resolve());
        testUser.mockReturnValue(Promise.reject(error));

        await put(event, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(testUser).toBeCalledWith(email, customerId);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when testCoupon throws error and no email in body', async () => {
        const noEmailEvent = {
          query: {
            accessToken,
          },
          cognitoPoolClaims: {
            stripe_customer_id: customerId,
          },
          body: {
            token,
            plan,
            promoCode,
            name,
            phone,
          },
        };
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.resolve());
        testCoupon.mockReturnValue(Promise.reject(error));

        await put(noEmailEvent, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(testUser).not.toBeCalled();
        expect(testCoupon).toBeCalledWith(promoCode);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when getCustomer throws error and no promo code in body', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.resolve());
        testUser.mockReturnValue(Promise.resolve());
        getCustomer.mockReturnValue(Promise.reject(error));

        await put(noPromoCodeEvent, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(testUser).toBeCalledWith(email, customerId);
        expect(testCoupon).not.toBeCalled();
        expect(getCustomer).toBeCalledWith(customerId);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when updateCustomer throws error', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.resolve());
        testUser.mockReturnValue(Promise.resolve());
        testCoupon.mockReturnValue(Promise.resolve());
        getCustomer.mockReturnValue(Promise.resolve({
          sourceId,
          subscriptionId,
          subscriptionItemId,
          couponId,
          plan,
        }));
        updateCustomer.mockReturnValue(Promise.reject(error));

        await put(event, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(testUser).toBeCalledWith(email, customerId);
        expect(testCoupon).toBeCalledWith(promoCode);
        expect(getCustomer).toBeCalledWith(customerId);
        expect(updateCustomer).toBeCalledWith(customerId, event.body);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when updateSubscription throws error and no token or email', async () => {
        const noTokenEmailEvent = {
          query: {
            accessToken,
          },
          cognitoPoolClaims: {
            stripe_customer_id: customerId,
          },
          body: {
            test: 'test',
            promoCode,
            plan,
            name,
            phone,
          },
        };
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.resolve());
        testCoupon.mockReturnValue(Promise.resolve());
        getCustomer.mockReturnValue(Promise.resolve({
          sourceId,
          subscriptionId,
          subscriptionItemId,
          couponId,
          plan,
        }));
        updateSubscription.mockReturnValue(Promise.reject(error));

        await put(noTokenEmailEvent, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(testUser).not.toBeCalled();
        expect(testCoupon).toBeCalledWith(promoCode);
        expect(getCustomer).toBeCalledWith(customerId);
        expect(updateCustomer).not.toBeCalled();
        expect(updateSubscription).toBeCalledWith(subscriptionId, noTokenEmailEvent.body);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when deleteDiscount throws error and null promo code', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.resolve());
        testUser.mockReturnValue(Promise.resolve());
        getCustomer.mockReturnValue(Promise.resolve({
          sourceId,
          subscriptionId,
          subscriptionItemId,
          couponId,
          plan,
        }));
        updateCustomer.mockReturnValue(Promise.resolve());
        deleteDiscount.mockReturnValue(Promise.reject(error));

        await put(noPromoCodeEvent, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(testUser).toBeCalledWith(email, customerId);
        expect(testCoupon).not.toBeCalled();
        expect(getCustomer).toBeCalledWith(customerId);
        expect(updateCustomer).toBeCalledWith(customerId, noPromoCodeEvent.body);
        expect(updateSubscription).not.toBeCalled();
        expect(deleteDiscount).toBeCalledWith(subscriptionId);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when deleteSubscription throws error, plan not null, and plan does not equal to current plan', async () => {
        const differentPlan = 'testDifferentPlan';
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.resolve());
        testUser.mockReturnValue(Promise.resolve());
        testCoupon.mockReturnValue(Promise.resolve());
        getCustomer.mockReturnValue(Promise.resolve({
          sourceId,
          subscriptionId,
          subscriptionItemId,
          couponId,
          plan: differentPlan,
        }));
        updateCustomer.mockReturnValue(Promise.resolve());
        deleteSubscription.mockReturnValue(Promise.reject(error));

        await put(event, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(testUser).toBeCalledWith(email, customerId);
        expect(testCoupon).toBeCalledWith(promoCode);
        expect(getCustomer).toBeCalledWith(customerId);
        expect(updateCustomer).toBeCalledWith(customerId, event.body);
        expect(updateSubscription).not.toBeCalled();
        expect(deleteDiscount).not.toBeCalled();
        expect(deleteSubscription).toBeCalledWith(subscriptionId);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when addSubscription throws error, plan not null, and plan does not equal to current plan', async () => {
        const differentPlan = 'testDifferentPlan';
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.resolve());
        testUser.mockReturnValue(Promise.resolve());
        testCoupon.mockReturnValue(Promise.resolve());
        getCustomer.mockReturnValue(Promise.resolve({
          sourceId,
          subscriptionId,
          subscriptionItemId,
          couponId,
          plan: differentPlan,
        }));
        updateCustomer.mockReturnValue(Promise.resolve());
        deleteSubscription.mockReturnValue(Promise.resolve());
        addSubscription.mockReturnValue(Promise.reject(error));

        await put(event, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(testUser).toBeCalledWith(email, customerId);
        expect(testCoupon).toBeCalledWith(promoCode);
        expect(getCustomer).toBeCalledWith(customerId);
        expect(updateCustomer).toBeCalledWith(customerId, event.body);
        expect(updateSubscription).not.toBeCalled();
        expect(deleteDiscount).not.toBeCalled();
        expect(deleteSubscription).toBeCalledWith(subscriptionId);
        expect(addSubscription).toBeCalledWith(customerId, event.body);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('correctly updates all stripe fields, except plan', async () => {
        testAuth.mockReturnValue(Promise.resolve());
        loadStripe.mockReturnValue(Promise.resolve());
        testUser.mockReturnValue(Promise.resolve());
        testCoupon.mockReturnValue(Promise.resolve());
        getCustomer.mockReturnValue(Promise.resolve({
          sourceId,
          subscriptionId,
          subscriptionItemId,
          couponId,
          plan,
        }));
        updateCustomer.mockReturnValue(Promise.resolve());
        deleteSubscription.mockReturnValue(Promise.resolve());
        addSubscription.mockReturnValue(Promise.resolve());

        await put(event, null, callback);

        expect(testAuth).toBeCalledWith(accessToken);
        expect(loadStripe).toBeCalled();
        expect(testUser).toBeCalledWith(email, customerId);
        expect(testCoupon).toBeCalledWith(promoCode);
        expect(getCustomer).toBeCalledWith(customerId);
        expect(updateCustomer).toBeCalledWith(customerId, event.body);
        expect(updateSubscription).toBeCalledWith(subscriptionId, event.body);
        expect(deleteDiscount).not.toBeCalled();
        expect(callback).toBeCalledWith(null, response);
        expect(global.console.log).toMatchSnapshot();
      });
    });
  });
});
