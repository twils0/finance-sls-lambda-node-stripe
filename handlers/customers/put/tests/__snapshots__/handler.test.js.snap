// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handlers users put correctly updates all stripe fields, except plan 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "testUser
",
      "test@test.com",
      "
",
      "testCustomerID",
      "
",
    ],
    Array [
      "testCoupon
",
      "testPromoCode",
      "
",
    ],
    Array [
      "getCustomer
",
      "testCustomerID",
      "
",
    ],
    Array [
      "getCustomer - success
",
      "testSourceID",
      "
",
      "testSubscriptionID",
      "
",
      "testCouponID",
      "
",
      "basic_plan",
      "
",
    ],
    Array [
      "updateCustomer
",
      "testCustomerID",
      "
",
      Object {
        "email": "test@test.com",
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": "testPromoCode",
        "token": Object {
          "test": "testToken",
        },
      },
      "
",
    ],
    Array [
      "updateSubscription
",
      "testSubscriptionID",
      "
",
      Object {
        "email": "test@test.com",
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": "testPromoCode",
        "token": Object {
          "test": "testToken",
        },
      },
      "
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users put fails and returns error, when addSubscription throws error, plan not null, and plan does not equal to current plan 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "testUser
",
      "test@test.com",
      "
",
      "testCustomerID",
      "
",
    ],
    Array [
      "testCoupon
",
      "testPromoCode",
      "
",
    ],
    Array [
      "getCustomer
",
      "testCustomerID",
      "
",
    ],
    Array [
      "getCustomer - success
",
      "testSourceID",
      "
",
      "testSubscriptionID",
      "
",
      "testCouponID",
      "
",
      "testDifferentPlan",
      "
",
    ],
    Array [
      "updateCustomer
",
      "testCustomerID",
      "
",
      Object {
        "email": "test@test.com",
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": "testPromoCode",
        "token": Object {
          "test": "testToken",
        },
      },
      "
",
    ],
    Array [
      "deleteSubscription
",
      "testSubscriptionID",
      "
",
      Object {
        "email": "test@test.com",
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": "testPromoCode",
        "token": Object {
          "test": "testToken",
        },
      },
      "
",
    ],
    Array [
      "addSubscription
",
      "testCustomerID",
      "
",
      Object {
        "email": "test@test.com",
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": "testPromoCode",
        "token": Object {
          "test": "testToken",
        },
      },
      "
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users put fails and returns error, when deleteDiscount throws error and null promo code 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "testUser
",
      "test@test.com",
      "
",
      "testCustomerID",
      "
",
    ],
    Array [
      "getCustomer
",
      "testCustomerID",
      "
",
    ],
    Array [
      "getCustomer - success
",
      "testSourceID",
      "
",
      "testSubscriptionID",
      "
",
      "testCouponID",
      "
",
      "basic_plan",
      "
",
    ],
    Array [
      "updateCustomer
",
      "testCustomerID",
      "
",
      Object {
        "email": "test@test.com",
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": null,
        "token": Object {
          "test": "testToken",
        },
      },
      "
",
    ],
    Array [
      "deleteDiscount
",
      "testSubscriptionID",
      "
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users put fails and returns error, when deleteSubscription throws error, plan not null, and plan does not equal to current plan 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "testUser
",
      "test@test.com",
      "
",
      "testCustomerID",
      "
",
    ],
    Array [
      "testCoupon
",
      "testPromoCode",
      "
",
    ],
    Array [
      "getCustomer
",
      "testCustomerID",
      "
",
    ],
    Array [
      "getCustomer - success
",
      "testSourceID",
      "
",
      "testSubscriptionID",
      "
",
      "testCouponID",
      "
",
      "testDifferentPlan",
      "
",
    ],
    Array [
      "updateCustomer
",
      "testCustomerID",
      "
",
      Object {
        "email": "test@test.com",
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": "testPromoCode",
        "token": Object {
          "test": "testToken",
        },
      },
      "
",
    ],
    Array [
      "deleteSubscription
",
      "testSubscriptionID",
      "
",
      Object {
        "email": "test@test.com",
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": "testPromoCode",
        "token": Object {
          "test": "testToken",
        },
      },
      "
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users put fails and returns error, when getCustomer throws error and no promo code in body 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "testUser
",
      "test@test.com",
      "
",
      "testCustomerID",
      "
",
    ],
    Array [
      "getCustomer
",
      "testCustomerID",
      "
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users put fails and returns error, when loadStripe throws error 1`] = `[MockFunction]`;

exports[`handlers users put fails and returns error, when testAuth throws error 1`] = `[MockFunction]`;

exports[`handlers users put fails and returns error, when testCoupon throws error and no email in body 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "testCoupon
",
      "testPromoCode",
      "
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users put fails and returns error, when testUser throws error 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "testUser
",
      "test@test.com",
      "
",
      "testCustomerID",
      "
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users put fails and returns error, when updateCustomer throws error 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "testUser
",
      "test@test.com",
      "
",
      "testCustomerID",
      "
",
    ],
    Array [
      "testCoupon
",
      "testPromoCode",
      "
",
    ],
    Array [
      "getCustomer
",
      "testCustomerID",
      "
",
    ],
    Array [
      "getCustomer - success
",
      "testSourceID",
      "
",
      "testSubscriptionID",
      "
",
      "testCouponID",
      "
",
      "basic_plan",
      "
",
    ],
    Array [
      "updateCustomer
",
      "testCustomerID",
      "
",
      Object {
        "email": "test@test.com",
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": "testPromoCode",
        "token": Object {
          "test": "testToken",
        },
      },
      "
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users put fails and returns error, when updateSubscription throws error and no token or email 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "testCoupon
",
      "testPromoCode",
      "
",
    ],
    Array [
      "getCustomer
",
      "testCustomerID",
      "
",
    ],
    Array [
      "getCustomer - success
",
      "testSourceID",
      "
",
      "testSubscriptionID",
      "
",
      "testCouponID",
      "
",
      "basic_plan",
      "
",
    ],
    Array [
      "updateSubscription
",
      "testSubscriptionID",
      "
",
      Object {
        "name": "testName",
        "phone": "239-555-0000",
        "plan": "basic_plan",
        "promoCode": "testPromoCode",
        "test": "test",
      },
      "
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
